datafusion-examples/examples/udaf_expr.rs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use datafusion::{
+    execution::{
+        config::SessionConfig,
+        context::{SessionContext, SessionState},
+    },
+    functions_aggregate::{expr_fn::first_value, register_all},
+};
+
+use datafusion_common::Result;
+use datafusion_expr::{col, AggregateUDFExprBuilder};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let ctx = SessionContext::new();
+    let config = SessionConfig::new();
+    let mut state = SessionState::new_with_config_rt(config, ctx.runtime_env());
+    let _ = register_all(&mut state);
+
+    let first_value_udaf = state.aggregate_functions().get("first_value").unwrap();
+    let first_value_builder = first_value_udaf
+        .call(vec![col("a")])
+        .order_by(vec![col("b")]);
```

Comment:
I think it would help to add comments to this example explaining what it was trying to show

Something like
```rust
// DataFusion also supports more advanced aggregate expressions, like
// `FIRST_VALUE(a ORDER BY b)`. To create such an expression using the expr_api
// you can use `AggregateUDFExprBuilder` trait as follows:
```

Also, I think it would actually be easier to find as a function in https://github.com/apache/datafusion/blob/main/datafusion-examples/examples/expr_api.rs rather than its own separate example